src/wasm/wasm-validator.cpp

```diff
@@ -2586,10 +2586,12 @@ void FunctionValidator::visitStructNew(StructNew* curr) {
                       "struct.new must have the right number of operands")) {
       // All the fields must have the proper type.
       for (Index i = 0; i < fields.size(); i++) {
-        shouldBeSubType(curr->operands[i]->type,
-                        fields[i].type,
-                        curr,
-                        "struct.new operand must have proper type");
+        if (!Type::isSubType(curr->operands[i]->type, fields[i].type)) {
+          info.fail("struct.new operand " + std::to_string(i) +
```

Comment:
What's the difference between the case you use `shouldBeSubType`'s error message and `info.fail`?

Reply:
Just that `shouldBeSubType` executes the arguments, so it would do wasted work to allocate a new string each time if I just added to `std::to_string` there.